packages/vite/src/node/plugins/html.ts

```diff
@@ -282,13 +297,67 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
               }
             }
           }
+          // for style tag and inline style use url(...) for assets references, also generate an import
+          const inlineStyle = node.props.find(
+            (prop) =>
+              prop.name === 'style' &&
+              prop.type === NodeTypes.ATTRIBUTE &&
+              prop.value
+          ) as AttributeNode
+          if (inlineStyle) {
+            if (!cssPlugin) {
+              cssPlugin = findCssPlugin(config)
+            }
+            const styleNode = inlineStyle.value!
+            resolveStyleAssetsTask.push(
+              Promise.resolve(
+                cssPlugin.transform!.call(this, styleNode.content, id + '.css')
+              ).then((compileResult) => {
+                if (compileResult) {
+                  const code =
+                    typeof compileResult === 'string'
+                      ? compileResult
+                      : compileResult.code || ''
+                  s.overwrite(
+                    styleNode.loc.start.offset,
+                    styleNode.loc.end.offset,
+                    `"${code}"`
+                  )
+                }
+              })
+            )
+          }
```

Comment:
Taking a step back, I think the more concise way of handling this is by using virtual modules, so we let the CSS file go through the entire plugin pipeline, instead of only the css plugin.

The flow would be similar to how Vite currently [handles inline scripts](https://github.com/vitejs/vite/blob/610d9f8381a44d4cb56766fb9beb8c547a8fb766/packages/vite/src/node/plugins/html.ts#L246-L258). Using [a map](https://github.com/vitejs/vite/blob/610d9f8381a44d4cb56766fb9beb8c547a8fb766/packages/vite/src/node/plugins/html.ts#L36-L39), and handle the virtual modules. This is only from my initial observation though, someone more well-versed in this area could give some thoughts on this too.

(Sorry for the late review. Been busy lately.)

Reply:
this is right, such as `<script>` tag in html. is good for the style tag.
But I had not idea for the inline style beause inline style had higher priority, can't use virtual file.
use `data:css;base64,` ?

Reply:
I believe inline style tags would have the same priority as external styles, and the precedence are checked by their ordering, so it would be fine.